runtime/interface.go

```diff
@@ -41,93 +43,110 @@ type Block struct {
 type ResolvedLocation = sema.ResolvedLocation
 type Identifier = ast.Identifier
 type Location = common.Location
+type AddressLocation = common.AddressLocation
 
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
```

Comment:
What about this one?

Reply:
yeah was not sure where this should live, probably break into two parts, part inside the accounts, part inside the runtime

Reply:
we might be able to do it through Contracts()

Reply:
This is an abstract method that allows the host environment choose how to resolve locations used by the Cadence program. For exampl, a Cadence declaration like `import ServiceAccount` specifies an `IdentifierLocation`, which the host environment through this call might resolve to an `AddressLocation` for a service account.

A function that lists contract names for an address doesn't fully cover that functionality.

Reply:
I still feel this should be on the cadence side, the only part of the current implementation on the FVM that needs the env is [these two lines](https://github.com/onflow/flow-go/blob/master/fvm/env.go#L213-L214), the rest is operations can stay on the cadence side.

Reply:
Different host environments might handle this differently and hard-coding this logic has made this hard to extend in the past: Initially this function didn't exists, and we had added it for multiple contracts per account. Having it allows us for example in the future to rewrite certain imports easily, e.g. rewrite something like `import FungibleToken from ServiceAccount` to `import 0x123`

Reply:
I see, I put it back then